tsconfig.json

```diff
@@ -1,14 +1,49 @@
 {
   "compilerOptions": {
-    "target": "es6",
+    "incremental": true,
+    "target": "es2017",
+    "outDir": "build/main",
+    "rootDir": "src",
+    "moduleResolution": "node",
     "module": "commonjs",
     "declaration": true,
-    "outDir": "./lib",
     "strict": true,
     "noImplicitAny": false,
     "esModuleInterop": true,
-    "allowSyntheticDefaultImports": true
+    "inlineSourceMap": true,
+    "resolveJsonModule": true /* Include modules imported with .json extension. */,
+
+    // "strict": true /* Enable all strict type-checking options. */,
+
+    /* Strict Type-Checking Options */
+    // "noImplicitAny": true /* Raise error on expressions and declarations with an implied 'any' type. */,
+    // "strictNullChecks": true /* Enable strict null checks. */,
+    // "strictFunctionTypes": true /* Enable strict checking of function types. */,
+    // "strictPropertyInitialization": true /* Enable strict checking of property initialization in classes. */,
+    // "noImplicitThis": true /* Raise error on 'this' expressions with an implied 'any' type. */,
+    // "alwaysStrict": true /* Parse in strict mode and emit "use strict" for each source file. */,
+
+    /* Additional Checks */
+    "noUnusedLocals": true /* Report errors on unused locals. */,
+    "noUnusedParameters": true /* Report errors on unused parameters. */,
+    "noImplicitReturns": true /* Report error when not all code paths in function return a value. */,
+    "noFallthroughCasesInSwitch": true /* Report errors for fallthrough cases in switch statement. */,
+
+    /* Debugging Options */
+    "traceResolution": false /* Report module resolution log messages. */,
+    "listEmittedFiles": false /* Print names of generated files part of the compilation. */,
+    "listFiles": false /* Print names of files part of the compilation. */,
+    "pretty": true /* Stylize errors and messages using color and context. */,
+
+    /* Experimental Options */
+    // "experimentalDecorators": true /* Enables experimental support for ES7 decorators. */,
+    // "emitDecoratorMetadata": true /* Enables experimental support for emitting type metadata for decorators. */,
+
+    "lib": ["es2017", "dom"],
+    "types": [],
+    "typeRoots": ["node_modules/@types", "src/types"]
   },
-  "include": ["src"],
-  "exclude": ["node_modules", "tests/*"]
+  "include": ["src/**/*.ts"],
```

Comment:
Is it necessary to add `*.ts` only? I don't know if it adds "more value" and it may be confusing if later we had .js / .json files in the repo and then it doesn't compile well because of this rule. I would prefer to keep `src` only.

Reply:
I agree, I will revert to `src` only.